sdk/typescript/src/module/executor.ts

```diff
@@ -67,3 +92,126 @@ export class Executor {
     return await builtObj[method](...Object.values(inputs))
   }
 }
+
+/**
+ * Interface Wrapper serves as dynaminc module binding so the module can
+ * call function of this interface.
+ * Because the actual interface implementation can come from any external modules,
+ * all resolution are done by API Call.
+ *
+ * @example
+ * ```ts
+ * interface Example {
+ *   foo: () => Promise<string>
+ * }
+ *
+ * class Test {
+ *   @func()
+ *   async callFoo(example: Example): Promise<string> {
+ *     // <- This example argument here is actually the Interface Wrapper, and `foo` will,
+ *     // directly execute the API call to reach the given implementation.
+ *     return example.foo()
+ *   }
+ * }
+ * ```
+ */
+class InterfaceWrapper {
```

Comment:
Lol this is neat. I spent ages trying to work out what was missing from this, since the go version of this *is so complex* - very neat :tada:

Reply:
Thanks :D